### PR TITLE
Refactor collector and app poller to use a timer

### PR DESCRIPTION
**Pull request by pradyutsarma**

### CODE REVIEW

File: src/autoscaler/metricscollector/collector/apppoller.go [hunk 1]
```diff
@@ -61,10 +61,11 @@ func (ap *appPoller) Stop() {
 func (ap *appPoller) startPollMetrics() {
 	for {
 		ap.pollMetric()
+		ap.timer.Reset(ap.pollInterval)
```
Comment (fraenkel) [inline src/autoscaler/metricscollector/collector/apppoller.go]: This is broken like I have discussed. Just stop the previous timer and create a new one.
Don't stop it in Stop().

File: src/autoscaler/metricscollector/collector/apppoller.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ type appPoller struct {
 	noaaConsumer noaa.NoaaConsumer
 	database     db.InstanceMetricsDB
 	pclock       clock.Clock
-	ticker       clock.Ticker
+	timer        clock.Timer
```
Comment (fraenkel) [inline src/autoscaler/metricscollector/collector/apppoller.go]: It should just live in the poll function.

**Comment by fraenkel** [comment]
@pradyutsarma Is this going to be fixed?

**Comment by pradyutsarma** [comment]
@fraenkel Yes. The code conflicts with the retry use case and hence, waiting for that to be merged.

**Comment by pradyutsarma** [comment]
Closing this. Re-created via this https://github.com/cloudfoundry-incubator/app-autoscaler/pull/144

**Pull request closed without merging**